### PR TITLE
Refs #7 - Implement Game/Forum Post Interface

### PR DESCRIPTION
**Pull request by nazar**
PR contains DraftJS GameMessage editor, with specific sample stylings for `shout`, `think` and `sing` - these samples are for demonstration purposes. The default status is `talk`

Only in-line styles have been implemented but block level styles (i.e. paragraph based elements such as headers) are to be implemented. One such example could be a `singing` block level element that takes a selection of several lines and turns it into a paragraph, with italics and centered text.... I'm sure we can think of more block level elements.

In character/ooc not implemented for this PR - we will revisit once character banners are in place (or before)

### CODE REVIEW

**Comment by mikelax** [comment]
Awesome @nazar. Can’t want to look at this.  
One comment though with regards to the styling. 

Even for the In character posting, I was thinking the default mode or text format would be that of “narrator”. 

Ie descriving what the character may be doing. If you look at my posts on BGG. You’ll see i narrate movements, actions, etc and then mix it up with character talking, thinking, etc.  

So I think the default node would be narrate and then  character speaking is just another option like singing, etc.

**Comment by nazar** [comment]
Yep - setting default/initial styles won't be an issue. We should be able to heavily customize the editor.

File: api/migrations/20171209140822_change_messages_message_to_json.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+exports.up = function(knex) {
+  return knex('game_messages')
+    .del()
```
Comment (mikelax) [inline api/migrations/20171209140822_change_messages_message_to_json.js:3]: scary!!, ha don't want to do this later on in the project dev.

File: client/src/components/Games/View/GameMessages.js [hunk 2]
```diff
@@ -44,7 +45,7 @@ class GameMessages extends Component {
     return <div className='game-messages'>
       {_.each(gameMessages).map(({ id, message }) => (
         <div key={id} className='game-message'>
-          {message}
+          <GameMessage message={message} readOnly={true} />
```
Comment (mikelax) [inline client/src/components/Games/View/GameMessages.js:48]: 👍

File: client/src/components/Games/components/GameMessage/GameMessage.js [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+function onToggleAction(entityKey) {
+  const editorState = this.state.editorState;
+  const currentContent = editorState.getCurrentContent();
+  const selection = editorState.getSelection();
+  const selectionEntity = getSelectionEntity(editorState);
+  const collapsed = selection.isCollapsed();
+
+  // many many many options here
```
Comment (mikelax) [inline client/src/components/Games/components/GameMessage/GameMessage.js:94]: 😨

File: client/src/components/Games/components/GameMessage/GameMessage.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    e.preventDefault();
+    this.props.onToggle(this.props.entity);
+  };
+}
+
+/*
+define individual styles as functional components
+*/
+
+const ThinkSpan = (props) => {
```
Comment (mikelax) [inline client/src/components/Games/components/GameMessage/GameMessage.js:186]: @nazar  do you want to document here or maybe in a readme section what is involved in adding a new style, ie. all the files that need to be updated?

**Review by mikelax — COMMENTED** [review]
@nazar  this is really great!
amazing work on getting this message component functional. i think maybe in a separate ticket we can work on the actual buttons and styles we want to launch with.,

File: client/src/components/Games/components/GameMessage/GameMessage.js (hunk 4 above)
Comment (nazar) [reply client/src/components/Games/components/GameMessage/GameMessage.js:186]: not just yet @mikelax as this component is likely change significantly - we'll definitely add documentation once we have a better idea of what components we'll need 📖

Comment (mikelax) [reply client/src/components/Games/components/GameMessage/GameMessage.js:186]: @nazar gotcha, makes sense. I left you a couple of messages on slack if you are around. i can help look at these two PRs and we can try to get them both merged.

**Review by mikelax — APPROVED** [review]
👍

**Pull request merged**